src/SonarQube.TeamBuild.PreProcessor/Interfaces/IPreprocessorObjectFactory.cs

```diff
@@ -37,7 +37,7 @@ public interface IPreprocessorObjectFactory
         /// <summary>
         /// Creates and returns the component to install the MSBuild targets
         /// </summary>
-        ITargetsInstaller CreateTargetInstaller();
+        ITargetsInstaller CreateTargetInstaller(ILogger logger);
```

Comment:
I would prefer the logger to be passed as a constructor argument to the factory, but did not wanted to make so many changes